keep/api/core/db.py

```diff
@@ -430,6 +435,94 @@ def get_workflows_that_should_run():
         return workflows_to_run
 
 
+def update_workflow_by_id(
+    id,
+    name,
+    tenant_id,
+    description,
+    interval,
+    workflow_raw,
+    is_disabled,
+    provisioned=False,
+    provisioned_file=None,
+    updated_by=None,
+):
+    with Session(engine, expire_on_commit=False):
+        existing_workflow = get_workflow(tenant_id, id)
+        if not existing_workflow:
+            raise ValueError("Workflow not found")
+        return update_workflow_by_workflow(
+            existing_workflow,
+            name=name,
+            description=description,
+            interval=interval,
+            workflow_raw=workflow_raw,
+            is_disabled=is_disabled,
+            provisioned=provisioned,
+            provisioned_file=provisioned_file,
+            updated_by=updated_by,
+        )
+
+
+def update_workflow_by_workflow(
+    existing_workflow: Workflow,
+    name,
+    description,
+    interval,
+    workflow_raw,
+    is_disabled,
+    provisioned=False,
+    provisioned_file=None,
+    updated_by=None,
+):
+    # In case the workflow name changed to empty string, keep the old name
+    if name != "":
+        name = name
+    else:
+        name = existing_workflow.name
+    with Session(engine, expire_on_commit=False) as session:
```

Comment:
it's another session you open right after using update_workflow_by_id, seems kinda redundant

Reply:
updated with `with existed_or_new_session(session) as session:`